docs/model_config.md

```diff
@@ -0,0 +1,364 @@
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+-->
+# FIL Backend Model Configuration
+Like all Triton backends, models deployed via the FIL backend make use of a
+specially laid-out "model repository" directory containing at least one
+serialized model and a `config.pbtxt` configuration file:
+```
+model_repository/
+├─ example/
+│  ├─ 1/
+│  │  ├─ model.json
+│  ├─ 2/
+│  │  ├─ model.json
+│  ├─ config.pbtxt
+```
+
+Documentation for general Triton configuration options is available in the
+[main Triton
+docs](https://github.com/triton-inference-server/server/blob/main/docs/user_guide/model_configuration.md#model-configuration),
+but here, we will review options specific to the FIL backend. For a more
+succinct overview, refer to the [FAQ notebook](https://nbviewer.org/github/triton-inference-server/fil_backend/blob/main/notebooks/faq/FAQs.ipynb), which includes guides for configuration under specific deployment scenarios.
+
+## Structure of configuration file
+A typical `config.pbtxt` file might look something like this:
+```
+backend: "fil"
+max_batch_size: 32768
+input [                                 
+ {  
+    name: "input__0"
+    data_type: TYPE_FP32
+    dims: [ 32 ]                    
+  } 
+]
+output [
+ {
+    name: "output__0"
+    data_type: TYPE_FP32
+    dims: [ 1 ]
+  }
+]
+instance_group [{ kind: KIND_AUTO }]
+parameters [
+  {
+    key: "model_type"
+    value: { string_value: "xgboost_json" }
+  },
+  {
+    key: "output_class"
+    value: { string_value: "true" }
+  }
+]
+
+dynamic_batching {}
+```
+Note that (as suggested by the file extension), this is a [Protobuf text file](https://developers.google.com/protocol-buffers/docs/text-format-spec)
+and should be formatted accordingly.
+
+## Specifying the backend
+If you wish to use the FIL backend, you must indicate this in the
+configuration file with the top-level `backend: "fil"` option. For
+information on models supported by the FIL backend, see [Model
+Support](https://developers.google.com/protocol-buffers/docs/text-format-spec)
+or the [FAQ
+notebook](https://nbviewer.org/github/triton-inference-server/fil_backend/blob/main/notebooks/faq/FAQs.ipynb#FAQ-1:-What-can-I-deploy-with-the-FIL-backend?)
+
+## Maximum Batch Size
+Because of the relatively quick execution speed of most tree models and the
+inherent parallelism of FIL, typically the only limitation on the maximum batch
+size is the size of Triton's CUDA memory pool [set at
+launch](https://nbviewer.org/github/triton-inference-server/fil_backend/blob/main/notebooks/faq/FAQs.ipynb#FAQ-4.2-How-do-I-increase-Triton's-device-memory-pool?).
+Nevertheless, you must specify *some* maximum batch size here, so setting it
+to whatever (large) value is consistent with your memory usage needs
+alongside other models should work fine:
+```
+max_batch_size: 1048576
```

Comment:
It should be safe for the vast majority of real users, but it is strictly dependent on hardware. We expect most real-life Triton deployments to be on e.g. H100/A100/T4/V100/similar, but Triton can also be used for edge deployments on e.g. Jetson boards. Setting the default to something compatible with a Jetson would leave throughput performance on the table for the bigger GPUs, and you even lose out a bit by targeting the lowest-end datacenter GPU memory. So, throughout the examples and docs, we instead encourage people to actually compute this value. Eventually, I'd like to look into a more sophisticated autoconfig setup, but that will have to come later.

In general, I think this value is a pretty reasonable compromise so that most people will see good performance out of the box when first testing the backend, but they can tweak it later based on their needs.